devlab/tests/test_verify_dst_del_tenant_resources.py

```diff
@@ -0,0 +1,239 @@
+             if 'deleted' in tenant and tenant['deleted'] is True]
+
+        self.dst_tenants = \
+            {ten.name: ten.id for
+             ten in self.dst_cloud.keystoneclient.tenants.list()}
+
+        dst_cinder_volumes = self.dst_cloud.cinderclient.volumes.list(
+            search_opts={'all_tenants': 0})
+
+        self.dst_volumes_admin = \
+            {vol.display_name: vol for vol in dst_cinder_volumes}
+
+        self.dst_vm_list = \
+            self.dst_cloud.novaclient.servers.list(
+                search_opts={'tenant_id': self.dst_tenants['admin']})
+
+    def _tenant_users_exist_on_dst(self, users_dict_list):
+        undeleted_users = []
+        dst_users_list = self.dst_cloud.keystoneclient.users.list()
+
+        for src_user_name in [user['name'] for user in users_dict_list]:
+            if src_user_name in dst_users_list:
+                undeleted_users.append(src_user_name)
+        return undeleted_users
+
+    def _net_is_attached_to_vm(self, src_net_name):
+        networks_names_lists = [vm.networks.keys() for vm in self.dst_vm_list]
+        network_names_list = sum(networks_names_lists, [])
+        return src_net_name in network_names_list
+
+    def _tenant_nets_exist_on_dst(self, tenant_networks):
+        dst_nets = self.dst_cloud.neutronclient.list_networks(
+            all_tenants=False)['networks']
+        nets_ids_list = []
+        for dst_net in dst_nets:
+            for src_net in tenant_networks:
+                if dst_net['name'] == src_net['name']:
+                    if not self._net_is_attached_to_vm(src_net['name']):
+                        nets_ids_list.append(dst_net['id'])
+        return nets_ids_list
+
+    def _tenant_vm_exists_on_dst(self, src_vm_list):
+        # according to current implementation,all tenant's vms
+        # should be moved tothe 'admin' account
+        src_vm_names = [vm['name'] for vm in src_vm_list]
+        migrated_vm_list = []
+        for dst_vm in self.dst_vm_list:
+            for src_vm in src_vm_list:
+                if src_vm['name'] == dst_vm.name:
+                    if self.dst_tenants['admin'] == dst_vm.tenant_id:
+                        migrated_vm_list.append(dst_vm.name)
+
+        non_migrated_vms = set(src_vm_names) ^ set(migrated_vm_list)
+        return non_migrated_vms
+
+    def _volume_is_attached_to_vm(self, vol_obj):
+        match_vm_id = 0
+        for attached_dev in vol_obj.attachments:
+            for dst_vm in self.dst_vm_list:
+                if dst_vm.id == attached_dev['server_id']:
+                    match_vm_id += 1
+        return len(vol_obj.attachments) == match_vm_id
+
+    def _tenant_volumes_exist_on_dst(self, src_volumes_list):
+        result_vol_ids = []
+        for volume_name, volume_obj in self.dst_volumes_admin.iteritems():
+            if volume_name in [vol['display_name'] for
+                               vol in src_volumes_list]:
```

Comment:
This code running in loop
```
if volume_name in [vol['display_name'] for vol in src_volumes_list]
```
I think You should rewrite part code in the following way
```
src_vol_list_by_disp_name = [vol['display_name'] for vol in src_volumes_list]
for volume_name, volume_obj in self.dst_volumes_admin.iteritems():
    if volume_name in src_vol_list_by_disp_name:
      .....
```